pkgs/by-name/vi/visual-hexdiff/package.nix

```diff
@@ -0,0 +1,50 @@
+{
+  stdenv,
+  lib,
+  fetchurl,
+  fetchpatch,
+  ncurses,
+}:
+stdenv.mkDerivation {
+  pname = "visual-hexdiff";
+  version = "0.0.53";
+
+  src = fetchurl {
+    url = "mirror://ubuntu/pool/universe/h/hexdiff/hexdiff_0.0.53.orig.tar.gz";
+    hash = "sha256-M1bmkW63pHlfl9zNWEq0EGN1rpVGo+BTUKM9ot4HWqo=";
+  };
+  postPatch = ''
+    # Fix compiler error that wants a string literal as format string for `wprintw`
+    substituteInPlace sel_file.c \
+      --replace-fail 'wprintw(win, txt_aide_fs[foo]);' 'wprintw(win, "%s", txt_aide_fs[foo]);'
+  '';
+
+  patches = [
+    # Some changes the debian/ubuntu developers made over the original source code
+    # See https://changelogs.ubuntu.com/changelogs/pool/universe/h/hexdiff/hexdiff_0.0.53-0ubuntu4/changelog
+    (fetchpatch {
+      url = "mirror://ubuntu/pool/universe/h/hexdiff/hexdiff_0.0.53-0ubuntu4.diff.gz";
+      sha256 = "sha256-X5ONNp9jeACxsulyowDQJ6REX6bty6L4in0/+rq8Wz4=";
+      decode = "gunzip --stdout";
+      name = "hexdiff_0.0.53-0ubuntu4.diff";
+      stripLen = 1;
+    })
+  ];
+
+  buildInputs = [ ncurses ];
+
+  preInstall = ''
+    mkdir -p $out/bin/
+  '';
+
+  makeFlags = [ "PREFIX=$(out)" ];
+
+  meta = with lib; {
+    description = "Visual hexadecimal difference editor";
+    homepage = "http://tboudet.free.fr/hexdiff/";
+    license = licenses.wtfpl;
+    maintainers = with maintainers; [ erictapen ];
+    mainProgram = "hexdiff";
+    platforms = platforms.all;
```

Comment:
Please fix or mark as broken for darwin
```
clang -o sel_file.o -c sel_file.c -g -DVERSION=\"0.0.53\" -DTRACE=0  -ansi
sel_file.c:22:7: error: conflicting types for 'strdup'
char *strdup(char *);
      ^
/nix/store/9l0nxr441n8qr0dk5n48m38dz4dswmg5-apple-sdk-10.12.2/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/string.h:117:7: note: previous declaration is here
char    *strdup(const char *__s1);
         ^
1 error generated.
```

Reply:
@FliegendeWurst I pushed a change that might fix the issue